bin/zowe-verify-authenticity.sh

```diff
@@ -0,0 +1,374 @@
+$SCRIPT.sh runtimePath hashPath
+
+   Parameter subsitutions:
+ 
+    Parm name       Sample value    Meaning
+    ---------       ------------    -------
+-r  runtimePath     /usr/lpp/zowe               Root directory of the executables used by Zowe at run time
+-h  hashPath        /usr/lpp/zowe/bin/internal  Directory of the hash key program
+-f  refPath         /usr/lpp/zowe/fingerprint   Directory of the reference hash key file
+-l  outputPath      ~/zowe/fingerprint          Output directory where log and work files like 
+                                                CustRuntimeHash.txt will be written.
+                                                This directory Will be created if it does not exist.
+
+EndOfUsage
+      exit 1
+      ;;
+  esac
+done
+
+# runtime path
+if [[ ! -n "$runtimePath" ]]
+then
+    # runtimePath was not set by -r
+    # Assume script was invoked in location ROOT_DIR/bin
+    # Allow relative or absolute path or tilde 
+    if [[ ! -d `dirname $0`/.. ]]
+    then
+        echo Error: Default runtimePath `dirname $0`/.. $notD
+        exit 1
+    fi
+    runtimePath=$(cd `dirname $0`;cd ..;pwd)
+fi
+
+
+# hash path
+if [[ ! -n "$hashPath" ]]
+then
+    if [[ ! -d `dirname $0`/../bin/internal ]]
+    then
+        echo Error: Default hashPath `dirname $0`/../bin/internal $notD
+        exit 1
+    fi 
+    hashPath=$(cd `dirname $0`;cd ../bin/internal;pwd)
+fi
+
+
+# ref path
+if [[ ! -n "$refPath" ]]
+then
+    if [[ ! -d `dirname $0`/../fingerprint ]]
+    then
+        echo Error: Default refPath `dirname $0`/../fingerprint $notD
+        exit 1
+    fi 
+    refPath=$(cd `dirname $0`;cd ../fingerprint || exit;pwd)
+fi
+
+# Create outputPath directory for log and other new files
+umask 0022
+if [[ ! -n "$outputPath" ]]
+then # create output directory in default location, suffixed 'fingerprint'
+    for dir in /global/zowe/log ~/zowe ${TMPDIR:-/tmp}
+    do
+        mkdir -p $dir/fingerprint 1> /dev/null 2>/dev/null 
+        if [[ $? -eq 0 ]]
+        then
+            outputPath=$dir/fingerprint
+            break
+        fi
+    done
+    if [[ ! -n "$outputPath" ]] # still failed to create a directory
+    then
+        echo Error: Cannot create default outputPath directory 
+        exit 1
+    fi
+else # create specified directory
+    mkdir -p $outputPath 1> /dev/null 2>/dev/null 
+    if [[ $? -ne 0 ]]
+    then
+        echo Error: Cannot create specified outputPath directory $outputPath 
+        exit 1
+    fi     
+fi
+outputPath=$(cd $outputPath;pwd)    # expand tilde and relative pathnames
+
+
+# is outputPath contained in runtime?
+echo $outputPath | grep ^$runtimePath  1> /dev/null 2> /dev/null
+if [[ $? -eq 0 ]]
+then
+    echo Error: outputPath $outputPath must not be in runtimePath $runtimePath
+    exit 1
+fi
+
+LOG_FILE=$outputPath/$SCRIPT.log
+touch $LOG_FILE
+if [[ $? -ne 0 ]]
+then
+    echo Error: Cannot write to $outputPath
+    exit 1
+fi
+
+echo Info: Logging to $LOG_FILE
```

Comment:
Use the logic in `setup-log-dir.sh` for this, rather than duplicating

Reply:
Now calls functions in `setup-log-dir.sh`